src/rpc/net.cpp

```diff
@@ -1014,52 +1015,107 @@ static RPCHelpMan sendmsgtopeer()
+                    }},
+                },
+            },
+            RPCResult{"for verbose = true", RPCResult::Type::OBJ_DYN, "", "json object with network type, new_table and tried_table as keys",
+                {
+                    {RPCResult::Type::ELISION, "", "The same output as verbose = false"},
+                    {
+                        RPCResult::Type::ARR, "table", "list of addresses in the address manager table ( new_table, tried_table )",
+                        {
+                            {RPCResult::Type::OBJ, "", "an address manager table entry", {
+                                {RPCResult::Type::STR, "address", "the address"},
+                                {RPCResult::Type::NUM, "services", "the services the node might support"},
+                                {RPCResult::Type::NUM, "bucket", "the address manager bucket the address is placed in"},
+                                {RPCResult::Type::NUM, "position", "the bucket position the address is placed in"},
+                                {RPCResult::Type::STR, "source", "the address that relayed the address to us"},
+                            }},
+                        },
+                    },
+                },
+            },
+        },
+        RPCExamples{
+            HelpExampleCli("getaddrmaninfo", "")
+            + HelpExampleCli("getaddrmaninfo", "true")
+            + HelpExampleRpc("getaddrmaninfo", "")
+            + HelpExampleRpc("getaddrmaninfo", "true")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue {
+            NodeContext& node = EnsureAnyNodeContext(request.context);
+            if (!node.addrman) {
+                throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Address manager functionality missing or disabled");
+            }
+
+            bool verbose = false;
+            if (!request.params[0].isNull()) {
+                verbose = request.params[0].get_bool();
+            }
+
+            UniValue ret(UniValue::VOBJ);
+            for (int n = 0; n < NET_MAX; ++n) {
+                enum Network network = static_cast<enum Network>(n);
+                if (network == NET_UNROUTABLE || network == NET_INTERNAL) continue;
+                UniValue obj(UniValue::VOBJ);
+                obj.pushKV("new", node.addrman->Size(network, true));
+                obj.pushKV("tried", node.addrman->Size(network, false));
+                obj.pushKV("total", node.addrman->Size(network));
+                ret.pushKV(GetNetworkName(network), obj);
+            }
+            UniValue obj(UniValue::VOBJ);
+            obj.pushKV("new", node.addrman->Size(std::nullopt, true));
+            obj.pushKV("tried", node.addrman->Size(std::nullopt, false));
+            obj.pushKV("total", node.addrman->Size());
+            ret.pushKV("all_networks", obj);
+
+            if (verbose) {
```

Comment:
We have `getmempoolinfo` that gives statistics about the mempool and `getrawmempool` that gives the contents of the mempool; wouldn't it make more sense to mirror that arrangement and have two separate RPCs here as well?